specs/credentialmanagement/index.src.html

```diff
@@ -927,18 +975,6 @@ <h4 id="request-credential">
       </ol>
     </li>
     <li>
-      Let <var>type</var> be lowest common ancestor interface of the types
-      referenced in <var>request</var>'s {{CredentialRequest/types}} property.
-
-      ISSUE(w3c/webappsec#289): This is terribly hand-wavey. The intent is
-      clear, but I need to do the work to walk through the list of DOMStrings
-      and convert them to interfaces and etc. Busywork for later.
```

Comment:
I don't think removing this is possible quite yet, as we don't have a good mechanism by which to deal with user interaction sanely. Consider a new credential type you instantiate: since you'll be storing it in DOM storage, you won't need to ask the user for permission to grab any of the data (nor will it be protected from any script on the page grabbing it, but that's a distinct concern). You will, however, need to ask user permission for `PasswordCredential` objects, for instance.

Which triggers first? Do we walk through the password store first, triggering user mediation, and fall back to the new credential type iff the user doesn't grant permission? If a `PasswordCredential` is available without mediation, is it alright to simply return it, or do we need to walk through the new credential types as well and mix the data somehow to allow the user to choose a result?

The current spec avoids this by rejecting the promise if multiple kinds of credentials are requested (e.g. anything that isn't a `LocallyStoredCredential` (which perhaps needs to be renamed `CredentialStoreCredential` or something equally verbose). Dropping that constraint needs to be accompanied by a solution to the problem. I don't think we have it yet.